io.missioncenter.MissionCenter.json

```diff
@@ -0,0 +1,169 @@
+{
+  "app-id": "io.missioncenter.MissionCenter",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "44",
+  "sdk": "org.gnome.Sdk",
+  "command": "missioncenter",
+  "finish-args": [
+    "--share=network",
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--device=all",
+    "--socket=wayland",
+    "--talk-name=org.freedesktop.Flatpak",
+    "--talk-name=org.gnome.Settings",
+    "--system-talk-name=org.freedesktop.systemd1",
+    "--system-talk-name=org.freedesktop.NetworkManager",
+    "--filesystem=xdg-data/flatpak/exports/share:ro",
+    "--filesystem=/var/lib/flatpak/exports/share:ro",
+    "--filesystem=xdg-data/flatpak/app:ro",
+    "--filesystem=/var/lib/flatpak/app:ro",
+    "--env=XDG_DATA_DIRS=/app/share:/usr/share:/usr/share/runtime/share:/run/host/user-share:/run/host/share:/var/lib/flatpak/exports/share:~/.local/share/flatpak/exports/share"
+  ],
+  "cleanup": [
+    "/include",
+    "/lib/pkgconfig",
+    "/man",
+    "/share/doc",
+    "/share/gtk-doc",
+    "/share/man",
+    "/share/pkgconfig",
+    "*.la",
+    "*.a"
+  ],
+  "modules": [
+    {
+      "name": "missioncenter",
+      "buildsystem": "simple",
+      "build-options": {
+        "append-path": "/run/build/missioncenter/bin",
+        "env": {
+          "RUST_BACKTRACE": "full",
+          "RUSTUP_HOME": "/run/build/missioncenter",
+          "CARGO_HOME": "/run/build/missioncenter/cargo"
+        }
+      },
+      "build-commands": [
+        "rustup default stable-x86_64-unknown-linux-gnu",
+        "cd subprojects/nvtop-* && find ../packagefiles -type f -name 'nvtop-*' -exec sh -c 'patch -p1 < {}' \\;",
+        "meson setup _flatpak_build -Dprefix=/app -Dbuildtype=release -Dflatpak=true -Dcargo_env=\"CARGO_HOME=$CARGO_HOME\"",
+        "ninja -C _flatpak_build install"
+      ],
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://gitlab.com/mission-center-devs/mission-center.git",
+          "tag": "v0.2.1",
+          "commit": "721d69a7a25f0903512d41e351d10db9db179928"
+        },
+        {
+          "type": "archive",
+          "url": "https://gitlab.com/mission-center-devs/build-tools/-/raw/main/rust-musl-1.70.0.tar.zst",
+          "sha256": "fcc101cd0af292d94e090b79517078c704201adff77817af0f654ca32f086f5b"
+        },
+        {
+          "type": "git",
+          "url": "https://gitlab.gnome.org/jwestman/blueprint-compiler",
```

Comment:
usually this is build as a separate moduel with a `cleanup: '*'` rule.

Reply:
Can you elaborate?
Normally blueprint-compiler is used as a meson subproject, and instead of letting meson grab and configure the sources I added it as a source that gets cloned into the subprojects folder and let meson handle it from there

Reply:
normal it is build in a flatpak manifest as a proper module, installed with a `cleanup: *` rule. That make it easy to remove it all from th build.

Reply:
I'm sorry, not trying to be difficult but I'm not sure what a "module installed with a `cleanup: *` rule is", can you point me to some documentation?
This might help me with replacing `remove-leftover-headers-and-libraries` module with a cleanup rule as well

Reply:
After some experimenting I understood what you meant, I think this last commit should do it